### PR TITLE
Add link to login for stanford only videos

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: app/javascript/src/controllers/media_tag_controller.js [hunk 1]
```diff
@@ -5,23 +5,50 @@ import mediaTagTokenWriter from '../modules/media_tag_token_writer.js'
...
+
     this.setupThumbnails()
+    this.validateMedia()
+  }
+
+  validateMedia(completeCallback) {
     const validators = this.mediaTagTargets
-      .map((mediaTag) => validator(mediaTag.dataset.authUrl, mediaTagTokenWriter(mediaTag)))
+      .map((mediaTag) => {
+        const mediaContext = { isRestricted: mediaTag.dataset.stanfordOnly === "true" || mediaTag.locationRestricted === "true" }
```
Comment (mjgiarlo) [inline app/javascript/src/controllers/media_tag_controller.js:22]: Is `locationRestricted` in the `dataset`?

File: app/javascript/src/controllers/media_tag_controller.js [hunk 2]
```diff
@@ -5,23 +5,50 @@ import mediaTagTokenWriter from '../modules/media_tag_token_writer.js'
...
+      values.forEach((result) => {
+        return this.afterValidate(result, completeCallback)})
     }).catch((err) => console.error(err))
   }
 
-  afterValidate(data) {
-    // TODO: handle stanford_restricted auth link
-    // TODO: handle is restricted
-    if(data.status === 'success') {
+  // NOTE: data.status can be a string or an array.
```
Comment (mjgiarlo) [inline app/javascript/src/controllers/media_tag_controller.js:31]: Should `data` be `result.authResponse`?

File: app/javascript/src/controllers/media_tag_controller.js [hunk 3]
```diff
@@ -5,23 +5,50 @@ import mediaTagTokenWriter from '../modules/media_tag_token_writer.js'
...
       this.initializeVideoJSPlayer()
     }
+
+    if(typeof(completeCallback) === 'function') {
+      completeCallback(result.authResponse);
+    }
+  }
+
+  isRestricted() {
+    const props = parentDiv.dataset
```
Comment (mjgiarlo) [inline app/javascript/src/controllers/media_tag_controller.js:50]: Where is `parentDiv` coming from?

Comment (jcoyne) [reply app/javascript/src/controllers/media_tag_controller.js:50]: Ooops, that's dead code. Removed it.

File: app/javascript/src/controllers/media_tag_controller.js (hunk 2 above)
Comment (jcoyne) [reply app/javascript/src/controllers/media_tag_controller.js:31]: Yes. Change made.

File: app/javascript/src/controllers/media_tag_controller.js (hunk 1 above)
Comment (jcoyne) [reply app/javascript/src/controllers/media_tag_controller.js:22]: Yes! Good eye.

**Review by mjgiarlo — APPROVED** [review]

File: app/javascript/src/controllers/media_tag_controller.js (hunk 3 above)
Comment (mjgiarlo) [reply app/javascript/src/controllers/media_tag_controller.js]: Cool.

**Pull request merged**